src/base/ceramicApiKit.ts

```diff
@@ -0,0 +1,197 @@
+                      confirmationsRequired: ${threshold}
+                  }
+                  clientMutationId: null
+              }
+          ) {
+              clientMutationId
+          }
+      }
+    `);
+
+
+    return Promise.resolve();
+  }
+
+  async getTransaction(safeTxHash: string): Promise<SafeMultisigTransactionResponse> {
+
+    try {
+      if (!this.composeClient) {
+        await this.connect();
+      }
+      const transactionIndex = await this.composeClient.executeQuery(`
+          query TransactionIndex {
+              transactionIndex(
+                  first: 1
+                  filters: { where: { safeTxHash: { equalTo: "${safeTxHash}" } } }
+              ) {
+                  edges {
+                      cursor
+                      node {
+                          id
+                          to
+                          data
+                          safe
+                          nonce
+                          value
+                          proposer
+                          operation
+                          safeTxHash
+                          signatures
+                          submissionDate
+                          transactionHash
+                          confirmationsRequired
+                      }
+                  }
+              }
+          }
+        `);
+
+      const confirmationsIndex: ConfirmationIndexData = await this.composeClient.executeQuery(`
+            query ConfirmationIndex {
+                confirmationIndex(
+                    first: 10
+                    filters: { where: { transactionHash: { equalTo: "${safeTxHash}" } } }
+                ) {
+                    edges {
+                        node {
+                            owner
+                            id
+                            signature
+                            signatureType
+                            submissionDate
+                            transactionHash
+                            confirmationType
+                        }
+                    }
+                }
+            }
+        `) as ConfirmationIndexData;
+      const confirmations = confirmationsIndex.data.confirmationIndex.edges.map((edge) => edge.node).filter((confirmation) => {
+        const { signature } = confirmation;
+        const r = signature.slice(0, 66);
+        const s = `0x${signature.slice(66, 130)}`;
+        let v = parseInt(signature.slice(130, 132), 16);
+        if (v !== 27 && v !== 28) {
+          v = 27;
+        }
+        // normalize signature
+        const normalizedSignature = r + s.slice(2) + (v).toString(16).padStart(2, '0');
+        return ethers.verifyMessage(ethers.getBytes(safeTxHash), normalizedSignature).toLowerCase() === confirmation.owner.toLowerCase();
+      })
+      return {
+        ...transactionIndex.data.transactionIndex.edges[0].node,
+        confirmations
+      } as SafeMultisigTransactionResponse;
```

Comment:
What to do if there are duplicates?

Reply:
and resolved, filtered here
https://github.com/helix-bridge/relayer/pull/30/files#diff-afa05a4785f248ff532b3a9c70ad3e9bbefcdd15491c95565b8e59e8546b187cR58